systemtest/produce_consume_group_test.go

```diff
@@ -0,0 +1,169 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemtest
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/elastic/apm-data/model"
+	apmqueue "github.com/elastic/apm-queue"
+	"github.com/elastic/apm-queue/codec/json"
+	"github.com/elastic/apm-queue/kafka"
+	"github.com/elastic/apm-queue/pubsublite"
+)
+
+func TestProduceConsumeMultipleGroups(t *testing.T) {
+	logger := NoLevelLogger(t, zap.ErrorLevel)
+
+	testCases := map[string]struct {
```

Comment:
This test case is not really testing the delivery. Delivery guarantees cover consumption by a consumer, not necessarily processing. I believe a better test would be:

- Start a consumer with a batch processor that fails processing records
- Produce 1 record
- Stop consumer after the record has been processed with an error
- Start a new consumer with a batch processor that succeeds processing records.
- Verify that the 1 record is:
  - processed for AtLeastOnceDelivery.
  - lost for AtMostOnceDelivery.

Reply:
Thanks for the details!

I finished the test but I'm having trouble testing it, I think I broke my env:

```
Error: Error when reading or editing Reservation: googleapi: Error 400: Attempted to delete a reservation with topics attached. The maximum amount of partitions which can be attached for this throughput is 0, but this change would leave 2 partitions attached. Please either add additional throughput_capacity or remove other topics.
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.ErrorInfo",
    "domain": "pubsublite.googleapis.com",
    "reason": "TOO_MANY_PARTITIONS_FOR_RESERVATION_THROUGHPUT"
  }
]
```

Reply:
strange, maybe you can try to run `terraform destroy` manually from `systemtest/tf/pubsublite` see if that helps cleaning up the dangling resources.

Reply:
That didn't work, I got the same error. I ended up logging into the cloud console and deleting the topics manually.